src/server/ws-routes/utilities.js

```diff
@@ -0,0 +1,38 @@
+const logger = require('../logger')
+
+const addNotification = (NotificationModel, UserModel, sendingNotificationUsername, recievingNotificationUsername, notificationType) => {
+	const notificationData = {
+		username: sendingNotificationUsername,
+		notificationType
+	}
+	const searchingCriteria = {
+		username: recievingNotificationUsername
+	}
+	const newNotification = new NotificationModel(notificationData)
+
+	return UserModel.findOne(searchingCriteria).exec()
+		.then(user => {
+			user.pendingNotifications.push(newNotification)
+			return user.save()
+		})
+		.then(() => newNotification)	
+}
+
+const sendMessageToSepcificUser = (socket, connetcions, recieverUsername, eventType, payload) => {
```

Comment:
1. `connetcions` -> `connections`
2. `message` word is confusing, since it may be interpreted as chat message, while you require eventType to be specified. What about `emitEventToUser`?